### PR TITLE
Issue 5704 - crash in sync_refresh_initial_content

### PR DESCRIPTION
**Pull request by tbordaz**
Bug description:
	If the last record of the changelog is not accessible
	then the session record is NULL. It crashes the server
	when it is dereferenced.
	I failed to reproduce it, including disabling/removing
	'cn=changelog' backend/mapping tree. So I guess it
	happens during rare dynamic.

Fix description:
	Return a failure when the session cookie is not
	initialized

relates: #5704

Reviewed by:

### CODE REVIEW

**Comment by abbra** [comment]
I keep seeing it crashing in IPA upgrade tests on Azure CI. `sync_cookie_get_change_info()` does an internal search for `cn=changelog` and it does not return an entry at all. This causes an error in generating a session cookie.

**Comment by tbordaz** [comment]
> I keep seeing it crashing in IPA upgrade tests on Azure CI. `sync_cookie_get_change_info()` does an internal search for `cn=changelog` and it does not return an entry at all. This causes an error in generating a session cookie.

Yes that is the scenario that this patch tries to eliminate. Is it still crashing with that patch ?

**Comment by abbra** [comment]
I haven't tried. Do you have a rawhide repo somewhere?

**Comment by tbordaz** [comment]
> I haven't tried. Do you have a rawhide repo somewhere?

Here is a build main+fix on [copr](https://download.copr.fedorainfracloud.org/results/tbordaz/F20buildDS389/fedora-rawhide-x86_64/05745109-389-ds-base/) repo

**Comment by abbra** [comment]
Thanks, so far I have not been able to reproduce a crash with this COPR build. But I haven't been able to reproduce it before locally as well...

File: ldap/servers/plugins/sync/sync_refresh.c [hunk 1]
```diff
@@ -101,6 +101,13 @@ sync_srch_refresh_pre_search(Slapi_PBlock *pb)
              */
             session_cookie = sync_cookie_create(pb, client_cookie);
             PR_ASSERT(session_cookie);
```
Comment (mreynolds389) [inline ldap/servers/plugins/sync/sync_refresh.c]: Should probably remove this assert since session_cookie does have a valid reason for being NULL.

Comment (tbordaz) [reply ldap/servers/plugins/sync/sync_refresh.c]: Correct, I removed it.

**Review by mreynolds389 — APPROVED** [review]
LGTM

**Pull request merged**